content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+---
+title: Mutating Webhook Good Practices
+description: >
+  Recommendations for writing mutating admission webhooks in Kubernetes.
+content_type: concept
+weight: 60
+---
+
+<!-- overview -->
+
+This page provides good practices and considerations when designing
+_mutating admission webhooks_ in Kubernetes. This information is intended for
+cluster operators who run your own admission webhook servers or third-party
+applications that modify your API requests.
+
+Before reading this page, ensure that you're familiar with the following
+concepts:
+
+* [Admission controllers](/docs/reference/access-authn-authz/admission-controllers/)
+* [Admission webhooks](/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)
+
+<!-- body -->
+
+## Importance of good webhook design {#why-good-webhook-design-matters}
+
+Mutating admission control occurs when any create, update, or delete request
+is sent to the Kubernetes API. These webhooks are often written to ensure that
+specific fields in object specifications exist or have specific allowed values.
+
+With every release, Kubernetes adds or modifies the API with new features,
+feature promotions to beta or stable status, and deprecations. Even stable
+Kubernetes APIs are likely might change. For example, the `Pod` API changed in
+v1.29 to add the
+[Sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/) feature.
+
+Webhooks that worked as expected with earlier versions of an API might not be
```

Comment:
```suggestion
It is rare or even not possible that the Kubernetes object that worked before will be broken by introduction of a new API, webhooks that worked as expected with earlier versions of an API might not be
```

Reply:
Trying to highlight the fact that in general upgrades are safe. The issue is that webhooks may be authored badly